pkg/validation/spec/swagger_test.go

```diff
@@ -126,3 +186,15 @@ func TestSwaggerSpec_Deserialize(t *testing.T) {
 		assert.EqualValues(t, actual, spec)
 	}
 }
+
+func TestSwaggerSpec_DeserializeYAML(t *testing.T) {
+	var actual Swagger
+	var nodes yaml.Node
+	err := yaml.Unmarshal([]byte(specYAML), &nodes)
+	assert.NoError(t, err)
+
+	err = nodes.Decode(&actual)
+	if assert.NoError(t, err) {
+		assert.EqualValues(t, actual, spec)
+	}
+}
```

Comment:
Not a comment for this PR, but we should add a roundtrip test in k/k for converting to YAML and back